docs/database-access-with-prisma-v2.md

```diff
@@ -8,565 +8,71 @@ This page explains how to use [Prisma](https://github.com/prisma/prisma) and the
+## What is `nexus-prisma`?
 
-Learn how to get started with Prisma [here](https://github.com/prisma/prisma2-docs/).
-
-### What is the `nexus-prisma` plugin?
-
-The [`nexus-prisma`](https://github.com/prisma/nexus-prisma) plugin is the glue between Photon and GraphQL Nexus. It generates CRUD building blocks for your Prisma models.
-
-![](https://imgur.com/dbEMHd5.png)
-
-When constructing your GraphQL schema with GraphQL Nexus, you build upon these building blocks and expose/customize them to your own API needs.
-
-### Generated CRUD building blocks
-
-Assume you have a `User` type in your Prisma schema:
-
-```groovy
-model User {
-  id    Int @id
-  name  String
-}
-```
-
-Define a prisma generator in your Prisma schema to generate the following building blocks for it:
-
-```groovy
-generator nexus_prisma {
-  provider = "nexus-prisma"
-}
-
-generator photon {
-  provider = "photonjs"
-}
-
-model User {
-  id    Int @id
-  name  String
-}
-```
-
-- **Queries**
-
-  - **`user(...): User!`**: Returns a single record
-  - **`users(...): [User!]!`**: Returns a list of records
-
-- **Mutations**
-
-  - **`createUser(...): User!`**: Creates a new record
-  - **`updateUser(...): User`**: Updates a record
-  - **`deleteUser(...): User`**: Deletes a record
-  - **`updateManyUsers(...): BatchPayload!`**: Updates many records in bulk
-  - **`deleteManyUsers(...): BatchPayload!`**: Deletes many records in bulk
-
-- [**GraphQL input types**](https://graphql.org/graphql-js/mutations-and-input-types/)
-  - **`UserCreateInput`**: Wraps all fields of the record
-  - **`UserUpdateInput`**: Wraps all fields of the record
-  - **`<Type>UserWhereInput`**: Provides filters for all fields of the record
-  - **`UserWhereUniqueInput`**: Provides filters for unique fields of the record
-  - **`UserUpdateManyMutationInput`**: Wraps fields that can be updated in bulk
-  - **`<Type>UserOrderByInput`**: Specifies ascending or descending orders by field
-
-> `UserCreateInput` and `UserUpdateInput` differ in the way relation fields are treated.
+[`nexus-prisma`](https://github.com/prisma/nexus-prisma) is a plugin for Nexus. It extends the Nexus DSL making it easy to expose Prisma models and operations against them in your GraphQL schema. Specifically the Nexus `t` is extended with `.model` and `.crud`. The former assists with exposing your Prisma Models as GraphQL Objects. The latter is valid within `Mutation` and `Query` Nexus type blocks and helps you expose operations over your Prisma Models including rich input types to express pagination, ordering, filtering and so forth.
 
 ## Examples
 
-Here's a minimal example for using `nexus-prisma`:
-
-**Prisma schema**:
+You can find a runnable `nexus-prisma` example in the [`nexus-prisma repo`](https://github.com/prisma/nexus-prisma/tree/next/example). The following are isolated snippets to give you a sense for the DSL.
```

Comment:
[`nexus-prisma repo`](https://github.com/prisma/nexus-prisma/tree/next/example) not sure if its purposely done but it's 404